### PR TITLE
completed task 6

### PR DESCRIPTION
**Pull request by GemboJones**
completed task 6 although feel I did a bit of a hack job for error handling in the model (fetchCommentsByArticleId), so feedback would be appreciated thanks!

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -127,4 +127,51 @@ describe('app', () => {
             })  
         })
     })
+    describe('GET /api/articles/:article_id/comments', () => {
+        test('200 : responds with an array of comments for the given article_id of which each comment should have the following properties; comment_id, votes, created_at, author, body, article_id', () => {
```
Comment (Augs0) [inline __tests__/app.test.js:131]: Nice job on this description :D

File: __tests__/app.test.js [hunk 2]
```diff
@@ -127,4 +127,51 @@ describe('app', () => {
...
+            return request(app)
+            .get('/api/articles/5/comments')
+            .expect(200)
+            .then((response) => {
+                const { comments } = response.body
+                expect(comments).toBeInstanceOf(Array)
+                comments.forEach((comment) => {
```
Comment (Augs0) [inline __tests__/app.test.js:140]: If you need to iterate over anything, be sure to check the length first

File: __tests__/app.test.js [hunk 3]
```diff
@@ -127,4 +127,51 @@ describe('app', () => {
...
+            })  
+        })
+        test('200 : responds with an empty array if the article_id exists but there are no comments for that article_id', () => {
+            return request(app)
+            .get('/api/articles/2/comments')
+            .expect(200)
+            .then((response) => {
+                const { comments } = response.body
+                expect(comments).toBeInstanceOf(Array)
+                expect(comments).toHaveLength(0)
```
Comment (Augs0) [inline __tests__/app.test.js:166]: To make sure we have handled this as expected, we probably want to test for a message. After all, if the article is not found, that also returns an empty array

File: __tests__/app.test.js [hunk 4]
```diff
@@ -127,4 +127,51 @@ describe('app', () => {
...
+        test('404 : responds with a 404 message when the path is valid but the article_id does not exist', () => {
+            return request(app)
+            .get('/api/articles/52/comments')
+            .expect(404)
+            .then((response) => {
+                const { msg } = response.body
+                expect(msg).toBe('not found')
+            })           
+        })
+    })
```
Comment (Augs0) [inline __tests__/app.test.js:187]: We also want a 400 for when a non numeric id is passed e.g. not-a-number

File: models/app.model.js [hunk 5]
```diff
@@ -31,4 +31,48 @@ const fetchArticles = () => {
     })
 }
 
-module.exports = { readTopics, fetchArticlesbyId, fetchArticles }
\ No newline at end of file
+const fetchCommentsByArticleId = (article_id) => {
+    return db.query(`SELECT articles.article_id, comment_id, comments.votes, comments.created_at, comments.author, comments.article_id 
+    FROM articles
```
Comment (Augs0) [inline models/app.model.js]: I can see the thinking behind what you have here and it's almost there! Remember, we want to select from the comments table and what foreign key connects the comments table and the articles table? The article_id, which we have access to 😉  how could we change this knowing this information?

File: models/app.model.js [hunk 6]
```diff
@@ -31,4 +31,48 @@ const fetchArticles = () => {
...
+            return db.query(`SELECT comments.comment_id, comments.votes, comments.created_at, comments.author, comments.body, comments.article_id 
+            FROM articles
+            LEFT JOIN comments ON articles.article_id = comments.article_id
+            WHERE comments.article_id = $1
+            ORDER BY created_at DESC;`, [article_id])
+            .then(({rows}) => {           
+                return rows
+            })
+        }
+    })
```
Comment (Augs0) [inline models/app.model.js:43]: A further hint for this one. As mentioned previously in my feedback, there are two scenarios where we get empty arrays: no comments for a valid article or a non-existent article. We will need to check the article exists in some way and once that is done - only once it's done - should we continue on. Do you remember how we can handle multiple promises?

File: models/app.model.js [hunk 7]
```diff
@@ -31,4 +31,48 @@ const fetchArticles = () => {
...
+                return rows
+            })
+        }
+    })     
+}
+
+module.exports = { readTopics, fetchArticlesbyId, fetchArticles, fetchCommentsByArticleId }
+
+
+/* const fetchCommentsByArticleId = (article_id) => {
```
Comment (Augs0) [inline models/app.model.js]: Please remove unused code

**Review by Augs0 — CHANGES_REQUESTED** [review]
Really excellent effort here Gemma! You are almost there on this one. The model needs a bit of reworking so take a look at my hints and see if you can figure it out. We're here if you need a hand!

**Comment by GemboJones** [comment]
hi @Augs0 if i work on these changes, how do i resubmit to you? Also i started working on task 7 on branch 7. will there be an issue with merging at all? Thanks!

**Pull request merged**